### PR TITLE
Improve agent starting and simultaneously restarting (common for Agent install) concurrency handling 

### PR DESCRIPTION
**Pull request by iglendd**
### What does this PR do?
Improve agent starting and simultaneously restarting, which is common for Agent install and upgrade, concurrency handling. Specifically it handles the following edge case:
1. [Process A] `datadogagent` service is starting in the end of Agent installation (no Agent services are running at this moment)
2. [Process B] Externally and concurrently (after Agent configuration change) `agent.exe restart-service` CLI command is started.
3. [Process B] `agent.exe restart-service` CLI command will get list of datadogagent dependent services to be stopped before stopping `datadogagent` service itself.
4. [Process A] `datadog-trace-agent` service (or other dependent services) is started by the still starting `datadogagent` service (from step no.1). 
5. [Process B] `agent.exe restart-service` CLI command should stop just started `datadog-trace-agent` service before moving on to stop `datadogagent` service itself.

Specifically:
* Collection of dependent services for the `datadogagent` service previously collected only `Running` dependent service. In some cases with concurrent execution stopping dependent services missed just started dependents services if they started AFTER dependent service collection,. The command would miss newly started dependent services and stopping `datadogagent` service would fail because a dependent service is still running. New code collects all dependent services to avoid this particular edge case. It does not solve all possible race condition but it eliminates a few of them.
* Previously timeout for service stop was 10 seconds and now it is increased to 30 seconds.

Both changes above are also similar to the Powershell's Restart-Service cmdlet logic.

### Motivation
We suspect that there is a small risk that certain small timing changes in Agent code may increase race conditions outlined above.

### Describe how to test/QA your changes
No need for manual test since key logic is tested in the accompanied unit tests

### CODE REVIEW

**Review by clarkb7 — APPROVED** [review]

File: releasenotes/notes/improve-agent-start-restart-concurrency-642420923b9ea275.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    Improve agent starting and simultaneously restarting concurrency handling.
```
Comment (clarkb7) [inline releasenotes/notes/improve-agent-start-restart-concurrency-642420923b9ea275.yaml:11]: This makes it sound like the change is about agent start/shutdown behavior, but it's really about the start/stop/restart agent commands. This seems like a better changelog for the follow PR you mention in the PR discription.

Comment (drichards-87) [inline releasenotes/notes/improve-agent-start-restart-concurrency-642420923b9ea275.yaml:11]:
```suggestion
    Improve concurrency handling for Agent start/simultaneous restart.
```

**Review by drichards-87 — APPROVED** [review]
Left you a minor suggestion from Docs and approved the PR.

File: releasenotes/notes/improve-agent-start-restart-concurrency-642420923b9ea275.yaml (hunk 1 above)
Comment (iglendd) [reply releasenotes/notes/improve-agent-start-restart-concurrency-642420923b9ea275.yaml]: thank you

File: pkg/util/winutil/service_test.go [hunk 2]
```diff
@@ -183,3 +188,309 @@ func TestListDependentServices(t *testing.T) {
...
+	StopService(s.mainSvcName)
+}
+
+func (s *StartStopServiceRaceConditionTestSuite) TestBasicRestartWhenNoSvcRunning() {
+	err := RestartService(s.mainSvcName)
+	assert.Nil(s.T(), err, "Main service should restar successfully: %v", err)
+
+	var status svc.Status
+	status, err = s.mainSvc.Query()
+	assert.Nil(s.T(), err, "Main service should restar successfully: %v", err)
```
[the remark this answers is not in the document]
Comment (iglendd) [reply pkg/util/winutil/service_test.go:311]: Will do, too bad I did not wait for you to review :), thank you

**Pull request merged**